Debian Linux 12 (amd64) build fails to compile gfx/drivers/gl3.c with `'struct <anonymous>' has no member named` errors

### Is there an existing issue for this?

- [x] This is a bug in RetroArch frontend
- [x] I have searched the existing issues

### Description

Platform: Debian Linux 12 (amd64)
Compiler: GNU GCC/G++

Multiple Compilation errors occurring on file `gfx/drivers/gl3.c`while compiling Retroarch (commit 3efaa0bb982e44d81a17621ae709f0576ed1fb73). The errors:

```
gfx/drivers/gl3.c: In function 'gl3_raster_font_draw_vertices':
gfx/drivers/gl3.c:991:24: error: 'struct <anonymous>' has no member named 'font_loc'
  991 |       if (gl->pipelines.font_loc.flat_ubo_vertex >= 0)
      |                        ^
gfx/drivers/gl3.c:992:36: error: 'struct <anonymous>' has no member named 'font_loc'
  992 |          glUniform4fv(gl->pipelines.font_loc.flat_ubo_vertex,
      |                                    ^
gfx/drivers/gl3.c: In function 'gl3_raster_font_setup_viewport':
gfx/drivers/gl3.c:1169:33: error: 'struct <anonymous>' has no member named 'font'
 1169 |       glUseProgram(gl->pipelines.font);
      |                                 ^
```

The build process was launched with the following commands:

```Bash
./configure --prefix=/path/to/prefix \
--disable-ffmpeg \
--disable-networkgamepad \
--disable-netplaydiscovery \
--disable-qt \
--disable-discord \
--disable-accessibility \
--disable-translate \
--disable-shaderpipeline \
--enable-builtinzlib \
--enable-sse

make clean
make
```

Attached are the configure script output and complete make output.

[configure.txt](https://github.com/user-attachments/files/23797917/configure.txt)
[make.txt](https://github.com/user-attachments/files/23797918/make.txt)

### Expected behavior

Build completes and a new Retroarch executable is produced.

### Steps to reproduce the bug

1. Checkout commit 3efaa0bb982e44d81a17621ae709f0576ed1fb73
2. Configure with command in the description above.
3. Invoke make

### Version/Commit

3efaa0bb982e44d81a17621ae709f0576ed1fb73

### Bisect Results

_No response_

### Present in the nightly version

I don't know

### Platform & operating system

Debian Linux 12 (amd64)

### Affected Cores

_No response_

### Environment information

* Linux KMS

### Relevant log output

```shell

```

--- a/gfx/drivers/gl3.c
+++ b/gfx/drivers/gl3.c
@@ -132,8 +132,8 @@ typedef struct gl3
    struct
    {
       GLuint alpha_blend;
-#ifdef HAVE_SHADERPIPELINE
       GLuint font;
+#ifdef HAVE_SHADERPIPELINE
       GLuint ribbon;
       GLuint ribbon_simple;
       GLuint snow_simple;
@@ -141,8 +141,8 @@ typedef struct gl3
       GLuint bokeh;
 #endif /* HAVE_SHADERPIPELINE */
       struct gl3_buffer_locations alpha_blend_loc;
-#ifdef HAVE_SHADERPIPELINE
       struct gl3_buffer_locations font_loc;
+#ifdef HAVE_SHADERPIPELINE
       struct gl3_buffer_locations ribbon_loc;
       struct gl3_buffer_locations ribbon_simple_loc;
       struct gl3_buffer_locations snow_simple_loc;
@@ -1661,12 +1661,12 @@ static void gl3_destroy_resources(gl3_t *gl)
       glDeleteProgram(gl->pipelines.alpha_blend);
       gl->pipelines.alpha_blend = 0;
    }
-#ifdef HAVE_SHADERPIPELINE
    if (gl->pipelines.font)
    {
       glDeleteProgram(gl->pipelines.font);
       gl->pipelines.font = 0;
    }
+#ifdef HAVE_SHADERPIPELINE
    if (gl->pipelines.ribbon)
    {
       glDeleteProgram(gl->pipelines.ribbon);
@@ -1987,11 +1987,11 @@ static bool gl3_init_pipelines(gl3_t *gl)
 #include "vulkan_shaders/alpha_blend.frag.inc"
       ;
 
-#ifdef HAVE_SHADERPIPELINE
    static const uint32_t font_frag[] =
 #include "vulkan_shaders/font.frag.inc"
       ;
 
+#ifdef HAVE_SHADERPIPELINE
    static const uint32_t pipeline_ribbon_vert[] =
 #include "vulkan_shaders/pipeline_ribbon.vert.inc"
       ;
@@ -2028,7 +2028,6 @@ static bool gl3_init_pipelines(gl3_t *gl)
    if (!gl->pipelines.alpha_blend)
       return false;
 
-#ifdef HAVE_SHADERPIPELINE
    if (!gl->pipelines.font)
       gl->pipelines.font = gl3_cross_compile_program(alpha_blend_vert, sizeof(alpha_blend_vert),
                                                       font_frag, sizeof(font_frag),
@@ -2036,6 +2035,7 @@ static bool gl3_init_pipelines(gl3_t *gl)
    if (!gl->pipelines.font)
       return false;
 
+#ifdef HAVE_SHADERPIPELINE
    if (!gl->pipelines.ribbon_simple)
       gl->pipelines.ribbon_simple = gl3_cross_compile_program(pipeline_ribbon_simple_vert, sizeof(pipeline_ribbon_simple_vert),
                                                                pipeline_ribbon_simple_frag, sizeof(pipeline_ribbon_simple_frag),

